boards/seeed/xiao_rp2040/doc/index.rst

```diff
@@ -0,0 +1,138 @@
+.. _xiao_rp2040:
+
+XIAO RP2040
+############
+
+Overview
+********
+
+The XIAO RP2040 is an IoT mini development board from Seeed Studio.
```

Comment:
I actually wonder if the board should just be `xiao` and have different SoCs, thoughts @tejlmand ? We really shouldn't be including the soc name in the name of the board anymore

Reply:
It may be worth keeping the scope of this card constrained to the addition of the new board. I'm a new zephyr developer and this feels like scope creep to me personally and outside of the scope of changes I would prefer to involve myself in (refactoring the other boards to fit this new setup being discussed, etc).

Reply:
This is not scope creep, we removed all socs from board names for hwmv2 because it's duplicated, in your PR here you would be building for `xiao_rp2040/rp2040` which doesn't really make any sense

Reply:
This board matches the current board folder structure of the existing xiao boards. Refactoring all of the xiao boards for some other new setup seems outside of the scope of a PR to add a new single board especially by a new contributor.

Reply:
> This is not scope creep, we removed all socs from board names for hwmv2 because it's duplicated, 

Well the board commercial name *is* XIAO RP2040 so I don't know about duplication. Whether it belongs to a series of boards that happen to have the same name should be mostly irrelevant. Unless we start refactoring all  "mimxrtevk" (is soc duplicated in `mimxrt1015_evk/mimxrt1015`?)  "stm32disco", "adafruit_feather", "nucleo", etc. (which we shouldn't) I really don't see why we would be asking this from @Timo614.

> in your PR here you would be building for `xiao_rp2040/rp2040` which doesn't really make any sense

Well, `west build -b xiao_rp2040 samples/hello_world/` seems fine to me :) If not then there are plenty of other "offenders" in tree which should (or shouldn't) be fixed first...

```
# these should be just "numaker pfm"?
numaker_pfm_m467/m467
numaker_pfm_m487/m487
# these should be lpcxpresso?
lpcxpresso11u68/lpc11u68
lpcxpresso51u68/lpc51u68
lpcxpresso55s06/lpc55s06
lpcxpresso55s16/lpc55s16
lpcxpresso55s28/lpc55s28
lpcxpresso55s36/lpc55s36
# intel ish?
intel_ish_5_4_1/intel_ish_5_4_1
intel_ish_5_6_0/intel_ish_5_6_0
intel_ish_5_8_0/intel_ish_5_8_0

```

Reply:
> This board matches the current board folder structure of the existing xiao boards

They should never have been merged, I did not see that